lib/auth/usertoken.go

```diff
@@ -452,6 +465,110 @@ func (s *Server) createRecoveryToken(ctx context.Context, username, tokenType st
 	return newToken, nil
 }
 
+// CreatePrivilegeToken implements AuthService.CreatePrivilegeToken.
+func (s *Server) CreatePrivilegeToken(ctx context.Context, req *proto.CreatePrivilegeTokenRequest) (types.UserToken, error) {
+	username, err := GetClientUsername(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	authPref, err := s.GetAuthPreference(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if !authPref.GetAllowLocalAuth() {
+		return nil, trace.AccessDenied("local auth needs to be enabled")
+	}
+
+	secondFactor := authPref.GetSecondFactor()
+	if secondFactor == constants.SecondFactorOff {
```

Comment:
It is interesting that we disallow SecondFactorOff, but we do allow the user to go through if they have no devices. Why one but not the other? Could leave a comment explaining why "off" is not allowed but empty devices are?